packages/namada.nix

```diff
@@ -35,15 +35,20 @@ pkgs.rustPlatform.buildRustPackage {
       "borsh-ext-1.2.0" = "sha256-nQadqyeAY0/gEMLBkpqtSm5D7kV+r3LVT/Cg2oTV7Ug=";
       "clru-0.5.0" = "sha256-/1NfKqcWGCyF3+f0v2CnuFmNjjKkzfvYcX+GzxLwz7s=";
       "ethbridge-bridge-contract-0.24.0" = "sha256-qs81bIWKk4oxh6nFWqAt4eBbPuIWF2a3ytUSjDJZWSU=";
+      "indexmap-2.2.4" = "sha256-uG8XMuoFt79cCZ8kqundQs++rqDLC/0ppiWToeAk5BE=";
       "index-set-0.8.0" = "sha256-oxJfQdKnYiW5VbMPuukVyDY5n8mys31hYNrJF89nXhY=";
       "ledger-namada-rs-0.0.1" = "sha256-qFL8LU7i5NAnMUhtrGykVfiYX1NodCNkZG07twyVrac=";
-      "masp_note_encryption-1.0.0" = "sha256-NwiosHTdzzny+L5VtOBaIa7wia/yRlfiz/8f0pAHUZk=";
+      "librocksdb-sys-0.16.0+8.10.0" = "sha256-ZcX2bpTDprcefo1ziyQ58GggX2D7NH17/zYvpbGSvhk=";
+      "masp_note_encryption-1.0.0" = "sha256-c3SiSCk5jK7eEXXXj2RWMig+HjOWztAeWoVZ+l+4KyU=";
       "sparse-merkle-tree-0.3.1-pre" = "sha256-B1ZEN4FZjV0x0Cqvx7AZjH9qhDMZYFPVJzg89dqWCv4=";
       "tiny-bip39-0.8.2" = "sha256-TU+7Vug3+M6Zxhy6Wln54Pxc9ES4EdFq5TvMOcAG+qA=";
       "tower-abci-0.11.1" = "sha256-KisZtsylvUymvV1TpDdGIiE7fSarcuD3I8oZ33BdKTU=";
-      "wasmer-2.3.0" = "sha256-Fd8ewAwslopjqUVoeHwSR/Zoh4Zm+Sdx8oksXmhLU20=";
+      "wasmer-2.3.0" = "sha256-lBipiaoYaEjJg9iZZrVhF1lUNH90QZ29xAHGBBhdujE=";
       "zcash_encoding-0.2.0" = "sha256-keuaoM/t1Q/+8JMemMMUuIo4g5I/EAoONFge+dyWGy0=";
     };
   };
+  preBuild = ''
+    export RUSTUP_TOOLCHAIN="1.77.1"
```

Comment:
Is this specific to Namada, or is this only for setting the Rust toolchain version?

If the latter, I believe you can instead do this by defining your own `rustPlatform` using https://github.com/oxalica/rust-overlay (which is already enabled in the flake) and then using that instead of `pkgs.rustPlatform`:

```
{ pkgs, namada-src }:
let
  rust = pkgs.rust-bin.stable."1.77.1".default;

  rustPlatform = nixpkgs.makeRustPlatform {
    cargo = rust;
    rustc = rust;
  };
in
rustPlatform.buildRustPackage {
```

Reply:
This was introduced for Namada specifically, after getting this error, https://github.com/informalsystems/cosmos.nix/actions/runs/8814425645/job/24194232530#step:6:2066, it seemed without the RUSTUP_TOOLCHAIN it was still using v1.75 which didn't have the fix.